external/storm-hdfs/src/main/java/org/apache/storm/hdfs/bolt/AbstractHdfsBolt.java

```diff
@@ -248,7 +248,9 @@ public void declareOutputFields(OutputFieldsDeclarer outputFieldsDeclarer) {
     @Override
     public void cleanup() {
         doRotationAndRemoveAllWriters();
-        this.rotationTimer.cancel();
+        if (this.rotationTimer != null) {
```

Comment:
This is the bugfix for the NPE. If the rotation policy is not timed, this field is null.